hbase/src/test/java/org/apache/metamodel/hbase/DeleteRowTest.java

```diff
@@ -0,0 +1,181 @@
+        try {
+            final HBaseTable existingTable = createAndAddTableToDatastore(TABLE_NAME, HBaseDataContext.FIELD_ID, CF_FOO,
+                    CF_BAR);
+            getUpdateCallback().deleteFrom(existingTable).execute();
+            fail("Should get an exception that the columnFamily doesn't exist.");
+        } catch (MetaModelException e) {
+            assertEquals("Key cannot be null", e.getMessage());
+        }
+    }
+
+    /**
+     * Creating a HBaseClient with the tableName null, should throw a exception
+     */
+    @Test
+    public void testCreatingTheHBaseClientWithTableNameNull() {
+        try {
+            new HBaseClient(getDataContext().getConnection()).deleteRow(null, new String("1"));
+            fail("Should get an exception that tableName is null");
+        } catch (IllegalArgumentException e) {
+            assertEquals("Can't delete a row without having tableName or rowKey", e.getMessage());
+        }
+    }
+
+    /**
+     * Creating a HBaseClient with the rowKey null, should throw a exception
+     */
+    @Test
+    public void testCreatingTheHBaseClientWithRowKeyNull() {
+        try {
+            new HBaseClient(getDataContext().getConnection()).deleteRow("tableName", null);
+            fail("Should get an exception that rowKey is null");
+        } catch (IllegalArgumentException e) {
+            assertEquals("Can't delete a row without having tableName or rowKey", e.getMessage());
+        }
+    }
+
+    /**
+     * Goodflow. Deleting a row, that doesn't exist, should not throw an exception
+     */
+    @Test
+    public void testDeletingNotExistingRow() {
+        try {
+            final HBaseTable existingTable = createAndAddTableToDatastore(TABLE_NAME, HBaseDataContext.FIELD_ID, CF_FOO,
+                    CF_BAR);
+
+            checkRows(false, false);
+            final HBaseRowDeletionBuilder rowDeletionBuilder = (HBaseRowDeletionBuilder) getUpdateCallback().deleteFrom(
+                    existingTable);
+            rowDeletionBuilder.setKey(RK_1);
+            rowDeletionBuilder.execute();
+            checkRows(false, false);
+        } catch (Exception e) {
+            fail("Should not get an exception that the row doesn't exist.");
+        }
+    }
+
+    /**
+     * Goodflow. Deleting a row, which has an empty rowKey value, should not throw an exception
+     */
+    @Test
+    public void testUsingAnEmptyRowKeyValue() {
+        try {
+            final HBaseTable existingTable = createAndAddTableToDatastore(TABLE_NAME, HBaseDataContext.FIELD_ID, CF_FOO,
+                    CF_BAR);
+
+            checkRows(false, false);
+            final HBaseRowDeletionBuilder rowDeletionBuilder = (HBaseRowDeletionBuilder) getUpdateCallback().deleteFrom(
+                    existingTable);
+            rowDeletionBuilder.setKey("");
+            rowDeletionBuilder.execute();
+            checkRows(false, false);
+        } catch (Exception e) {
```

Comment:
Same like earlier, don't catch the exception but have the method throw `IOException` instead.